### PR TITLE
Js 8

### PR DESCRIPTION
**Pull request by rat6666**

### CODE REVIEW

File: todo_task/src/actions/eventsHandlerService.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+export class EventsHandlerService {
+  constructor(rootNode, taskRepository, renderPage) {
+    this.rootNode = rootNode;
+    this.taskAction = new TaskAction(taskRepository, renderPage);
+  }
+
+  onAction() {
+    this.rootNode.addEventListener('click', () => {
+      switch (event.target.classList[0]) {
```
Comment (ClainAtomi) [inline todo_task/src/actions/eventsHandlerService.js:18]: easy to fail code, if someone will add another class into target element and break class sequence. Better to check all classList and find exact class from your list.

File: todo_task/src/actions/taskAction.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+export class TaskAction {
+  constructor(tasksRepository, onChange) {
+    this.tasksRepository = tasksRepository;
+    this.onChange = onChange;
+    this.storage = {
```
Comment (ClainAtomi) [inline todo_task/src/actions/taskAction.js:5]: too mixed variable, add more meaningfull names and split if possible.
when i read `valueDown` i can't understand where this variable can be applicable in code.

File: todo_task/src/actions/taskAction.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  onTaskRemove(taskID) {
+    this.tasksRepository.removeTask(taskID).then(() => {
+      this.onDone(this.storage.valueDelete);
+    });
+  }
+
+  onTaskPosting() {
+    const newTask = document.querySelector('[class="input-add"]').value;
```
Comment (ClainAtomi) [inline todo_task/src/actions/taskAction.js:27]: `'[class="input-add"]'` move such strings into config object

Comment (ClainAtomi) [reply todo_task/src/actions/taskAction.js:27]: no magic strings rule :)

File: todo_task/src/render/headerRenderer.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+export class HeaderRenderer {
+  constructor() {
+    this.taskPosition = {
```
Comment (ClainAtomi) [inline todo_task/src/render/headerRenderer.js:3]: better to move such enums into other files and just import them everywhere

File: todo_task/src/script.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+);
+const eventsHandlerService = new EventsHandlerService(
+  rootNode,
+  taskRepository,
+  pageService.renderPage
+);
+
+pageService.renderPage(footerText).then(() => {
+  eventsHandlerService.onAction();
+});
```
Comment (ClainAtomi) [inline todo_task/src/script.js:42]: good job here :)

File: todo_task/src/render/headerRenderer.js (hunk 4 above)
Comment (rat6666) [reply todo_task/src/render/headerRenderer.js:3]: done

File: todo_task/src/actions/taskAction.js (hunk 3 above)
Comment (rat6666) [reply todo_task/src/actions/taskAction.js:22]: done

File: todo_task/src/script.js (hunk 5 above)
Comment (rat6666) [reply todo_task/src/script.js:39]: :)

File: todo_task/src/actions/taskAction.js (hunk 2 above)
Comment (rat6666) [reply todo_task/src/actions/taskAction.js]: done (i hope)

File: todo_task/src/actions/eventsHandlerService.js (hunk 1 above)
Comment (rat6666) [reply todo_task/src/actions/eventsHandlerService.js]: done

**Pull request merged**